packages/playback-core/src/media-tracks.ts

```diff
@@ -131,3 +136,20 @@ export function setupMediaTracks(
   // NOTE: Since this is only relevant for hls, using destroying event (CJP).
   hls.once(Hls.Events.DESTROYING, removeAllMediaTracks);
 }
+
+function fetchMetadata(el: HTMLMediaElement, metadataUrl: string) {
+  fetch(metadataUrl)
+    .then((resp) => resp.json())
+    .then((json) => {
+      // Attempt to set metadata on the custom element
+      if ('metadata' in el) {
+        (el as any).metadata = json;
```

Comment:
don't use the public element API internally here. `fetchMetadata` will have to be a private method because we want to set a private property.